components/access-management/Dockerfile

```diff
@@ -1,19 +1,20 @@
-FROM golang:1.12.4
+ARG GOLANG_VERSION=1.12.4
+FROM golang:${GOLANG_VERSION} as builder
 
-RUN mkdir -p /opt/kubeflow
-RUN mkdir -p $GOPATH/src/github.com/kubeflow/kubeflow/components/access-management
-WORKDIR $GOPATH/src/github.com/kubeflow/kubeflow/components/access-management
-
-ENV PATH /go/bin:/usr/local/go/bin:$PATH
-# use go modules
-ENV GO111MODULE=on
+WORKDIR /workspace
 
 COPY . .
 RUN go mod download
-RUN go build -gcflags 'all=-N -l' -o /opt/kubeflow/access-management main.go
+RUN go build -gcflags 'all=-N -l' -o access-management main.go
+RUN chmod a+rx access-management
 
-RUN chmod a+rx /opt/kubeflow/access-management
+# Use distroless as minimal base image to package the manager binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+FROM gcr.io/distroless/static:latest as serve
```

Comment:
Why static? It looks like the goland example is using base
https://github.com/GoogleContainerTools/distroless/blob/master/examples/go/Dockerfile

Reply:
updated